witness/src/Dscp/Witness/Launcher/Mode.hs

```diff
@@ -60,21 +68,31 @@ type WitnessWorkMode ctx m =
     , HasLens' ctx MempoolVar
     , HasLens' ctx SDActions
     , HasLens' ctx (KeyResources WitnessNode)
-
+    , HasLens' ctx TxRelayInput
+    , HasLens' ctx TxRelayPipe
     )
 
 ---------------------------------------------------------------------
 -- WorkMode implementation
 ---------------------------------------------------------------------
 
+newtype TxRelayInput = TxRelayInput { getTxRelayInput :: STM.TQueue GTxWitnessed }
+newtype TxRelayPipe  = TxRelayPipe  { getTxRelayPipe  :: STM.TQueue GTxWitnessed }
+
+newTxRelayInput :: MonadIO m => m TxRelayInput
+newTxRelayInput = TxRelayInput <$> atomically STM.newTQueue
+
+newTxRelayPipe :: MonadIO m => m TxRelayPipe
+newTxRelayPipe = TxRelayPipe <$> atomically STM.newTQueue
+
 -- | Context is resources plus some runtime variables.
 data WitnessContext = WitnessContext
-    { _wcParams    :: !WitnessParams
-      -- ^ Parameters witness was started with.
-    , _wcResources :: !WitnessResources
-      -- ^ Resources, allocated from params.
-    , _wcMempool   :: !MempoolVar
-    , _wcSDActions :: !SDActions
+    { _wcParams       :: !WitnessParams     -- ^ Parameters witness was started with.
+    , _wcResources    :: !WitnessResources  -- ^ Resources, allocated from params.
+    , _wcMempool      :: !MempoolVar
```

Comment:
Mempool access is not required anymore actually, could you remove it?

Reply:
I use mempool later, so - no.

Reply:
Lol, right